insonmnia/hub/eth.go

```diff
@@ -112,7 +112,8 @@ func (e *eth) CheckDealExists(id string) (bool, error) {
 		return false, err
 	}
 
-	idOK := deal.GetSupplierID() == util.PubKeyToAddr(e.key.PublicKey)
+	// NOTE: May GetSupplierID return common.Address?
+	idOK := deal.GetSupplierID() == util.PubKeyToAddr(e.key.PublicKey).Hex()
```

Comment:
`util.EqualAddresses()`?

Reply:
See the Note. If GetSupplierID was common.Address